### PR TITLE
Resolve Simulation Failures

### PR DESCRIPTION
**Pull request by JoongunPark**
## Summary
- Fixed a sporadic issue where All-Reduce callback was not being triggered.
- Broadcast is not implemented but occurs in some traces (Resnet-50). Future implementation required.

## Test Plan
I've tested with the latest ASTRA-Sim in Ubuntu 22.04. 
The traces are from Resnet-50.

### CODE REVIEW

File: astra-sim/workload/Workload.cc [hunk 1]
```diff
@@ -252,19 +252,24 @@ void Workload::issue_comm(shared_ptr<Chakra::ETFeederNode> node) {
       collective_comm_node_id_map[fp->my_id] = node->id();
       collective_comm_wrapper_map[fp->my_id] = fp;
       fp->set_notifier(this, EventType::CollectiveCommunicationFinished);
+
```
Comment (TaekyungHeo) [inline astra-sim/workload/Workload.cc:255]: Let's avoid any unnecessary changes like blank lines.

File: astra-sim/system/Sys.cc [hunk 2]
```diff
@@ -1386,6 +1386,10 @@ void Sys::proceed_to_next_vnet_baseline(StreamBaseline* stream) {
     return;
   }
   stream->steps_finished++;
+  // This is hot fix for random failures of simulation.
+  // TODO: Need to find a better way for negative queue id occurrence.
+  if (stream->phases_to_go.front().queue_id < 0)
```
Comment (TaekyungHeo) [inline astra-sim/system/Sys.cc:1391]: Why does it happen? Can we track down the reason?

File: astra-sim/workload/Workload.cc (hunk 1 above)
Comment (JoongunPark) [reply astra-sim/workload/Workload.cc:255]: I put that blank lines because the other if-statement did that in the same way. 
Do you think we need to remove all those as well?

File: astra-sim/system/Sys.cc (hunk 2 above)
Comment (JoongunPark) [reply astra-sim/system/Sys.cc:1391]: For now, it seems initialization issue. Let me keep track that. 
But for now, it can effectively remove the error.
